src/discord-cluster-manager/bot.py

```diff
@@ -0,0 +1,125 @@
+            await self.add_cog(MiscCog(self))
+
+            guild_id = (
+                DISCORD_CLUSTER_STAGING_ID
+                if not self.debug_mode
+                else DISCORD_DEBUG_CLUSTER_STAGING_ID
+            )
+
+            if guild_id:
+                guild = discord.Object(id=int(guild_id))
+                self.tree.clear_commands(guild=guild)
+                logger.info("Cleared existing commands")
+                self.tree.copy_global_to(guild=guild)
+                await self.tree.sync(guild=guild)
+                commands = await self.tree.fetch_commands(guild=guild)
+                logger.info(f"Synced commands: {[cmd.name for cmd in commands]}")
+        except Exception as e:
+            logger.error(f"Failed to sync commands: {e}")
+
+    async def on_ready(self):
+        logger.info(f"Logged in as {self.user}")
+        for guild in self.guilds:
+            try:
+                if self.debug_mode:
+                    await guild.me.edit(nick="Cluster Bot (Staging)")
+                else:
+                    await guild.me.edit(nick="Cluster Bot")
+            except Exception as e:
+                logger.warning(f"Failed to update nickname in guild {guild.name}: {e}")
+
+    async def create_thread(
+        self, interaction: discord.Interaction, gpu_name: str, job_name: str
+    ) -> discord.Thread:
+        thread = await interaction.channel.create_thread(
+            name=f"{job_name} ({gpu_name}) - {datetime.now().strftime('%Y-%m-%d %H:%M')}",
+            type=discord.ChannelType.public_thread,
+            auto_archive_duration=1440,
+        )
+        return thread
+
+    async def send_chunked_message(
+        self, channel, content: str, code_block: bool = True
+    ):
+        """
+        Send a long message in chunks to avoid Discord's message length limit
+
+        Args:
+            channel: The discord channel/thread to send to
+            content: The content to send
+            code_block: Whether to wrap the content in code blocks
+        """
+        chunk_size = 1900  # Leave room for code block syntax
+        chunks = [
+            content[i : i + chunk_size] for i in range(0, len(content), chunk_size)
+        ]
+
+        for i, chunk in enumerate(chunks):
+            if code_block:
+                await channel.send(
+                    f"```\nOutput (part {i+1}/{len(chunks)}):\n{chunk}\n```"
+                )
+            else:
+                await channel.send(chunk)
+
+
+def main():
+    init_environment()
+
+    parser = argparse.ArgumentParser(description="Run the Discord Cluster Bot")
+    parser.add_argument(
+        "--debug", action="store_true", help="Run in debug/staging mode"
+    )
+    args = parser.parse_args()
+
+    logger.info("Starting bot...")
+    token = DISCORD_DEBUG_TOKEN if args.debug else DISCORD_TOKEN
+
+    if args.debug and not token:
+        raise ValueError("DISCORD_DEBUG_TOKEN not found")
```

Comment:
Maybe change to the following so that it catches a missing token outside of debug mode? It leaves it up to the user to figure out which token was missing (DISCORD_TOKEN vs. DISCORD_DEBUG_TOKEN) but that seems OK to me (the user will be able to figure it out).

```suggestion
    if not token:
        raise ValueError("Discord token not found")
```

Reply:
Just noticed the code in consts.py is already checking for DISCORD_TOKEN. I think the existing check is fine.